A `maybe-uninitialized` warning with GCC 13 in `median.hpp`

This shows up in a SciPy CI job with `-Werror` when moving the build from GCC 11.2.0 to GCC 13.3.0: 

```
  [834/1264] Compiling C++ object scipy/stats/_stats_pythran.cpython-312-x86_64-linux-gnu.so.p/meson-generated__stats_pythran.cpp.o
  In file included from scipy/stats/_stats_pythran.cpython-312-x86_64-linux-gnu.so.p/_stats_pythran.cpp:101:
  ../../../../../../tmp/pip-build-env-0__sc8gh/overlay/lib/python3.12/site-packages/pythran/pythonic/numpy/median.hpp: In function ‘decltype ((declval<T>() + 1.0e+0)) {anonymous}::pythonic::numpy::median(const {anonymous}::pythonic::types::ndarray<T, pS>&, {anonymous}::pythonic::types::none_type) [with T = double; pS = {anonymous}::pythonic::types::pshape<long int>]’:
  ../../../../../../tmp/pip-build-env-0__sc8gh/overlay/lib/python3.12/site-packages/pythran/pythonic/numpy/median.hpp:68:7: warning: ‘*<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
     68 |     T t0 = tmp[n / 2];
        |       ^~
  ../../../../../../tmp/pip-build-env-0__sc8gh/overlay/lib/python3.12/site-packages/pythran/pythonic/numpy/median.hpp: In function ‘decltype ((declval<T>() + 1.0e+0)) {anonymous}::pythonic::numpy::median(const {anonymous}::pythonic::types::ndarray<T, pS>&, {anonymous}::pythonic::types::none_type) [with T = double; pS = {anonymous}::pythonic::types::array_base<long int, 1, {anonymous}::pythonic::types::tuple_version>]’:
  ../../../../../../tmp/pip-build-env-0__sc8gh/overlay/lib/python3.12/site-packages/pythran/pythonic/numpy/median.hpp:68:7: warning: ‘*<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
     68 |     T t0 = tmp[n / 2];
        |       ^~
```

Not urgent since I can just silence the warning, but would be nice to resolve.

--- a/pythran/pythonic/numpy/median.hpp
+++ b/pythran/pythonic/numpy/median.hpp
@@ -65,7 +65,14 @@ namespace numpy
     T *tmp = utils::allocate<T>(n);
     std::copy(arr.buffer, arr.buffer + n, tmp);
     std::nth_element(tmp, tmp + n / 2, tmp + n, ndarray::comparator<T>{});
+#if defined(_GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
     T t0 = tmp[n / 2];
+#if defined(_GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
     if (n % 2 == 1) {
       utils::deallocate(tmp);
       return t0;

